### ISSUE TITLE
Error with spectrogram

### ISSUE THREAD

**Issue by alexarje** [fourMs/MGT-python#190]
Getting an error `AttributeError: 'AxesSubplot' object has no attribute 'get_array'` when creating a spectrogram. The spectrogram shows up, though.

![Screenshot from 2021-01-15 11-06-40](https://user-images.githubusercontent.com/114316/104711614-e1dfa600-5721-11eb-90a4-7355051094e5.png)

**Comment by balintlaczko** [comment]
Thanks. Looks mysterious, as if the colorbar would not understand the axes object it should connect to. Found some other people complaining about this, without an obvious solution ([1](https://askpythonquestions.com/2020/11/17/adding-colorbar-to-seaborn-bubble-plot/), [2](https://github.com/matplotlib/matplotlib/issues/14892), [3](https://rasterio.groups.io/g/main/topic/rio_plot_show_with_colorbar/73065089?p=), [4](https://stackoverflow.com/questions/60587415/adding-colorbar-to-scatter-plot)).

I couldn't actually reproduce this problem, but found that there _is_ another problem (opened #191 for that one). 

@alexarje can you check what version of 'matplotlib' you are using? I'm on 3.3.3 on both OS-es. Also, try updating matplotlib. If it fixes it, I will just add matplotlib 3.3.3 as minimum requirement.

**Comment by alexarje** [comment]
I had matplotlib 3.1.1. Have upgraded it to 3.3.3 now, but getting some error...

**Comment by balintlaczko** [comment]
OK, thanks for the attempt anyway! Did this happen in the intro notebook? Does this happen with any kind of video files?
Also, I will test it with `.m4v` videos, but if you could provide a little excerpt from yours it would also help!

**Comment by TeunKrikke** [comment]
Hi,

I came across this page whilst looking for a solution for the same problem but with a different library. I have found a small work around that might help but looking at the code it is something that needs to be done in the library. The issue seems that img in the colorbar is of the type AxesSubplot which in matplotlib is not compatible with the ScalarMappable that colorbar expects. I don't know how to solve it but as said have a work around for this.

`cmap = mpl.cm.magma # this is the colormap used to display the spectrogram
norm = mpl.colors.Normalize(vmin=-80, vmax=0) # these are the min and max values from the spectrogram

fig.colorbar(mpl.cm.ScalarMappable(norm=norm, cmap=cmap),
             ax=ax, format="%+2.f dB") # this bit of code displays the colorbar (instead of line 72 as in the screenshot of the top post)`
